### PR TITLE
Healium is too powerful we need to nerf it!

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Healium was too stronk, nerfed it to behave like a good gas.
Added coughing and euphoria mood if breathed in small quantities to alert the player.
Sleep if it is 3 moles in the air for 3 to 5 seconds
Sleep you if is more than 6 moles and it heals you while knocked down.
Lowered the amount of healing done

Edit: It doesnt knock you out, it makes you sleep now, since the knockdown effect was so strong that people wouldn't wake up

## Why It's Good For The Game
fix and nerf

## Changelog
:cl:
add: Added a minimum amount for healium knock down, added a warning when traces are present
tweak: Remove knockdown, added sleep (less intense effect)
refractor: refractored sleeping debuff to take in consideration the TRAIT_SLEEPIMMUNE
refractor: moved the healing effects to the reagent
del: removed oxyloss from the reagent effect
tweak: lowered the amount of healing done by the healium gas
/:cl:

### CODE REVIEW

**Comment by trerri** [comment]
What if healium instead of K.O.ing you just paralyzed you and got you horizontal so you could still scream for help?

**Comment by LemonInTheDark** [comment]
I will read this in the not now, you will be spared for your great crimes against le balance

File: code/modules/surgery/organs/lungs.dm
Comment (UnlimitedVanillaWorks) [inline code/modules/surgery/organs/lungs.dm:342]:
```suggestion
				to_chat(H, "<span class='alert'>Your head starts spinning and your lungs burn!</span>")
```

**Comment by CRITAWAKETS** [comment]
what in the goddamn are those healing numbers, this shit heals twice as good as atropine and can be stacked

File: code/modules/surgery/organs/lungs.dm
Comment (Ghilker) [reply code/modules/surgery/organs/lungs.dm:342]: Ah thanks!

**Comment by Ghilker** [comment]
It's an hard gas to make and it knock you out to get the effect, so you can't wake up on your own and leave you open and helpless

**Comment by cacogen** [comment]
Soothed. Sedated.

**Comment by ExcessiveUseOfCobblestone** [comment]
what the hell is healium and why is there a gas that is healing people

**Comment by Ghilker** [comment]
is a new gas and it heals people because it doesn't know it can't heal people so it does it anyway

**Comment by ExcessiveUseOfCobblestone** [comment]
what the heck...

**Comment by Barhandar** [comment]
Basically it's a meme gas that is better N2O for surgeries.

For all of the 0.5 people on all servers combined that use anaesthetic.

File: code/modules/surgery/organs/lungs.dm
Comment (ATH1909) [inline code/modules/surgery/organs/lungs.dm:350]: if you're no longer using the healium reagent for this, it might be best to remove it (the healium reagent), so that botanists can't roll it as a strange seeds chem or something.

Comment (Ghilker) [reply code/modules/surgery/organs/lungs.dm:350]: mmm i could remove it true, unless there is a way to make the reagent sleep instead of knockdown

Comment (ExcessiveUseOfCobblestone) [reply code/modules/surgery/organs/lungs.dm:350]: reagents can do anything to the person a gas could do.

File: code/datums/status_effects/debuffs.dm
Comment (Rohesie) [inline code/datums/status_effects/debuffs.dm:166]:
```suggestion
```
No need for this check, we are already making it not tick when this happens.

File: code/modules/surgery/organs/lungs.dm
Comment (Rohesie) [reply code/modules/surgery/organs/lungs.dm:350]: Put this back into the reagent code.

File: code/modules/surgery/organs/lungs.dm
Comment (Rohesie) [inline code/modules/surgery/organs/lungs.dm:355]: These should be moved to the reagent's `on_mob_life()`

File: code/modules/reagents/chemistry/reagents/other_reagents.dm
Comment (Rohesie) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1376]: Make a child of the sleeping status effect. You want a perma-sleep one here, until `on_mob_end_metabolize()` is called, not the regular timed sleep.

**Comment by Ghilker** [comment]
don't merge yet, need to resolve the perma sleep

File: code/modules/mob/living/status_procs.dm
Comment (LemonInTheDark) [inline code/modules/mob/living/status_procs.dm:305]: Check what might cause ignore_stun to be passed in as true, if it's used at all pass it in to IsSleeping() and if it's true, ignore the trait check.

File: code/modules/mob/living/status_procs.dm
Comment (LemonInTheDark) [inline code/modules/mob/living/status_procs.dm:294]: Ready for suffering? has_status_effect() does not in fact return a boolean yes or no as the name implies, it returns the trait itself. so &&'ng it with a bool check will cause undesired behavior. Maybe 
```suggestion
  if(!HAS_TRAIT(src, TRAIT_SLEEPIMMUNE))
    return has_status_effect(STATUS_EFFECT_SLEEPING)
```

File: code/datums/status_effects/debuffs.dm
Comment (LemonInTheDark) [inline code/datums/status_effects/debuffs.dm:163]: If anything uses that override you will need to reconsider this whole block, @Rohesie thoughts?

File: code/modules/reagents/chemistry/reagents/other_reagents.dm
Comment (Rohesie) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1387]:
```suggestion
	L.adjustOxyLoss(-5, FALSE)
	L.adjustFireLoss(-7, FALSE)
	L.adjustToxLoss(-7, FALSE)
	L.adjustBruteLoss(-10, FALSE)
```

File: code/modules/mob/living/status_procs.dm
Comment (Rohesie) [reply code/modules/mob/living/status_procs.dm:305]: Yep, these checks need to go, all of them. The proc's parameter as well.

File: code/datums/status_effects/debuffs.dm
Comment (Rohesie) [reply code/datums/status_effects/debuffs.dm:164]: Could you rephrase the issue?

Comment (LemonInTheDark) [reply code/datums/status_effects/debuffs.dm:164]: If anything uses the override bool on the "make sleep" procs, this refactor will change behavior

**Review by Rohesie — CHANGES_REQUESTED** [review]
In `medicine_reagents.dm`, `M.Sleeping(100, 0, TRUE)` should be `M.Sleeping(100, FALSE)`

File: code/datums/status_effects/debuffs.dm
Comment (Rohesie) [reply code/datums/status_effects/debuffs.dm:164]: Addressed through discord.
In short, the `ignore_canstun` parameter on the sleeping procs is being removed, in favor of having the sleep immunity trait respected.
If things want to force unconsciousness onto anything then they shouldn't do it through the sleeping proc, but using the knockedout trait.

**Review by Rohesie — DISMISSED** [review]

**Comment by Ghilker** [comment]
Has cobby requested, i've lowered the amount of healing done by the gas to a more sane amounts

File: code/modules/reagents/chemistry/reagents/other_reagents.dm
Comment (Rohesie) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1384]:
```suggestion
	. = ..()
	if(L.stat < UNCONSCIOUS)
		return
	L.adjustOxyLoss(-2, FALSE)
```
This is my only balance concern. Don't think this should heal a conscious mob.
If you neutralize it via another gas then you need to eschew the benefits.

Comment (Ghilker) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:1384]: I think that removing the healing from a conscious mob will remove the reason to combine it with other gases, for the oxy i could just remove it is not a big deal

Comment (Rohesie) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:1384]: What do you think, @ExcessiveUseOfCobblestone ?
I don't have too much of a strong opinion here besides wanting to avoid regen sources stacking at potentially no cost.

Comment (Ghilker) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: as cobby told me, i removed the oxyloss but i kept the other interactions, we'll see how it's going to play out

**Comment by Rohesie** [comment]
Update the changelog and description after the changes.
Once that is done, this is ready for merge.

**Comment by Ghilker** [comment]
should be all, if it's missing something else please tell me

**Comment by LemonInTheDark** [comment]
Your changelog mentions both removing the reagent and refactoring it, please update it to better reflect the state of things

**Comment by Ghilker** [comment]
F I changed it now

**Pull request merged**
